Update Readme.md (#6)
DevExpress-Examples/xaf-how-to-display-an-enumeration-property-as-a-drop-down-box-with-check-boxes: Readme.md

<!-- default badges list -->
[![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/E689)
[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
<!-- default badges end -->

> # XAF - How to Display an Enumeration Property as a Drop-down Box with Check Boxes

Reviewer: Looks like the WinForms example creates a drop-down check list, while the Blazor example creates a check list box (no drop-down window). So I would remove the words "Drop-down" from the description. 

```suggestion
# XAF - How to Display an Enumeration Property as a Check List
```

> This example demonstrates how to create a custom Property Editor that displays an enumeration property as a drop-down box with check boxes, if the enumeration type has `FlagsAttribute` that allows to treat this enumeration as a set of flags.

Reviewer: It's a pretty long sentence and I broke it down into two. 

The word "FlagsAttribute" needs an article. First of all, you don't refer to the type in general, you refer to an individual attribute instance assigned to a type. So you need an indefinite article... like when you would say "add a FlagsAttribute attribute to the type". (You kind of add a new instance or create something new.) But yes, then you want to remove repetition of the word "attribute". That leaves you with "a FlagsAttribute"  that reads like "a Flags attribute". 

```suggestion
This example demonstrates how to create a custom Property Editor that displays an enumeration property as a check list. The enumeration type must be declared with a `FlagsAttribute` (allows to treat this enumeration as a set of flags).
```

![image](https://user-images.githubusercontent.com/14300209/234846358-4435cbbe-130d-410b-9958-fb8450a11480.png)

## Implementation Details

> Since XAF does not have a built-in Property Editor for [enumeration properties](https://docs.devexpress.com/eXpressAppFramework/113552/business-model-design-orm/data-types-supported-by-built-in-editors/enumeration-properties) that can store several values at once,  you need to implement a custom Property Editor. It takes data property value as input and displays it in the UI with the help of a custom visual control. In an XAF Windows Forms application, the control is based on [`CheckedComboBoxEdit`](https://docs.devexpress.com/WindowsForms/DevExpress.XtraEditors.CheckedComboBoxEdit) editor from the XtraEditors Library. In an XAF ASP.NET Core Blazor application, it is based on the [DxListBox](https://docs.devexpress.com/Blazor/DevExpress.Blazor.DxListBox-2) control from the Blazor Library.

Reviewer: We don't really use "Xtra..." names anymore. And the fact that the control comes from the "editors library" is not important either. A long time ago, a customer could buy an XtraEditors Library license separately. That's no longer possible. It's just one of our WinForms controls and no additional details are necessary. 

And since we won't mention the XtraEditors Library, no need for a parallel construct in the sentence about Blazor. 

```suggestion
XAF does not have a built-in Property Editor for [enumeration properties](https://docs.devexpress.com/eXpressAppFramework/113552/business-model-design-orm/data-types-supported-by-built-in-editors/enumeration-properties) that can store several values at once.  You need to implement a custom Property Editor that obtains the data property value and displays it in the UI with the help of a custom visual control. In an XAF Windows Forms application, the control is [`CheckedComboBoxEdit`](https://docs.devexpress.com/WindowsForms/DevExpress.XtraEditors.CheckedComboBoxEdit). In an XAF ASP.NET Core Blazor application, the control is [DxListBox](https://docs.devexpress.com/Blazor/DevExpress.Blazor.DxListBox-2).
```
